factcast-factus/src/main/java/org/factcast/factus/Factus.java

```diff
@@ -118,4 +119,10 @@ default void publish(@NonNull List<EventObject> eventPojos) {
 
   // conversion
   Fact toFact(@NonNull EventObject e);
+
+  LockedOnSpecs<? extends Projection> withLockOn(@NonNull FactSpec spec);
```

Comment:
the type parameter is pretty useless, should be cleaned up by replacing Locked by
LockedOnSpec and LockedOnProjection at some point...